HEC.FDA.Model/hydraulics/HydraulicDataset.cs

```diff
@@ -65,21 +65,34 @@ public bool LoadFromXML(XElement xElement)
 
         #endregion
         #region Methods
-        public List<UncertainPairedData> GetGraphicalStageFrequency(string pointShapefileFilePath, string parentDirectory)
+
+        /// <summary>
+        /// Translates a point shapefile into points, reprojects those points according to the specified projection, samples each hydraulic profile at those points, and returns the results as a list of UncertainPairedData objects.
+        /// </summary>
+        public List<UncertainPairedData> GetGraphicalStageFrequency(string pointShapefileFilePath, string parentDirectory, Projection studyProjection)
         {
-            List<UncertainPairedData> ret = new List<UncertainPairedData>();
-            PointFeatureLayer indexPoint = new PointFeatureLayer("ThisNameIsNotUsedForAnythingHere", pointShapefileFilePath);
+            List<UncertainPairedData> ret = new();
+            PointFeatureLayer indexPoint = new("ThisNameIsNotUsedForAnythingHere", pointShapefileFilePath);
             if (!indexPoint.SourceFileExists)
             {
                 return null;
             }
-            PointMs indexPoints = new PointMs(indexPoint.Points().Select(p => p.PointM()));
+            PointMs indexPoints = new(indexPoint.Points().Select(p => p.PointM()));
+            
+            //reproject the point to the study projection. For GetWSE, all the reprojection is done in the Inventory class. We aren't using an inventory here, so we need to do it manually.
+            Projection pointProj = RASHelper.GetVectorProjection(indexPoint);
```

Comment:
This is a substantial change from what we did before. Seems correct to me, but could be the change?

Reply:
No. Red herring. I think this is fine.